azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -539,6 +541,11 @@ private void prepareAndStartServer() throws Exception {
     configureRoutes();
     startWebMetrics();
 
+    // start ContainerMetrics only if containerized execution is enabled
+    if (isContainerizedDispatchMethodEnabled()) {
+      startContainerMetrics();
+    }
```

Comment:
Since ContainerMetrics will be used as constructor members of a few more classes (e.g. AbstractExecutorManagerAdapter), I assume DummyContainerMetricsImpl could be useful in future unit tests. Following the same logic, if containerized dispatch method is not enabled, DummyContainerMetricsImpl could be injected into AbstractExecutorManagerAdapter to avoid generating unnecessary container metrics.